### PR TITLE
[ACTION] Added new action

### PR DESCRIPTION
**Pull request by jcortes**
### <samp>🤖 Generated by Copilot at dcd3814</samp>

> _A new action for Scrape-It.Cloud_
> _Lets you scrape endpoints with a loud_
> _Click of a button_
> _But you need some mutton_
> _To parse and validate what's allowed_

### CODE REVIEW

File: components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import app from "../../scrape_it_cloud.app.mjs";
+import utils from "../../common/utils.mjs";
+
+export default {
+  key: "scrape_it_cloud-start-scraping",
+  name: "Start Scraping",
+  description: "Initiate the scraping process for a specific endpoint. [See the documentation here](https://scrape-it.cloud/docs/api-features/basic-request).",
```
Comment (dylburger) [inline components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:7]:
```suggestion
  description: "Initiate the scraping process for a specific endpoint. [See the documentation](https://scrape-it.cloud/docs/api-features/basic-request)."
```

Comment (dylburger) [reply components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:7]: Removed "here" from the link text as per Google's Developer docs style guide.

File: components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  type: "action",
+  version: "0.0.1",
+  props: {
+    app,
+    url: {
+      type: "string",
+      label: "URL",
+      description: "Specify the URL of the web page you would like to scrape.",
```
Comment (dylburger) [inline components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:15]:
```suggestion
  description: "Specify the URL of the web page you want to scrape."
```

Comment (dylburger) [reply components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:15]: Changed "would like to" to "want to" as per Google's Developer docs style guide.

File: components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    },
+    headers: {
+      type: "object",
+      label: "Headers",
+      description: "Specify custom HTTP headers to be passed to the request. For example, in order to override the User-Agent header, use `User-Agent` as a key and `Teapot` as a value.",
```
Comment (dylburger) [inline components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:20]:
```suggestion
  description: "Specify custom HTTP headers to pass to the request. For example, to override the User-Agent header, use `User-Agent` as a key and `Teapot` as a value."
```

Comment (dylburger) [reply components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:20]: Changed "in order to" to "to" as per Google's Developer docs style guide.

File: components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      optional: true,
+    },
+    blockResources: {
+      type: "boolean",
+      label: "Block Resources",
+      description: "By default Scrape-it.cloud do not block images and CSS in the scraped page. To speed up requests and block images and CSS set this parameter value to true.",
```
Comment (dylburger) [inline components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:26]:
```suggestion
  description: "By default, Scrape-it.cloud doesn't block images and CSS in the scraped page. To speed up requests and block images and CSS, set this parameter value to true."
```

Comment (dylburger) [reply components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:26]: Changed "do not" to "doesn't" and removed unnecessary hyphen as per Google's Developer docs style guide.

File: components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      optional: true,
+    },
+    blockUrls: {
+      type: "string[]",
+      label: "Block URLs",
+      description: "If you want to block some resources except images and CSS, for example, analytics scripts you can add part of the urls to be blocked. Follow [Blocking URLs](https://scrape-it.cloud/docs/api-features/blocking-urls) page to get more information.",
```
Comment (dylburger) [inline components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:32]:
```suggestion
  description: "If you want to block some resources besides images and CSS, for example, analytics scripts, you can add part of the URLs to be blocked. Follow the [Blocking URLs](https://scrape-it.cloud/docs/api-features/blocking-urls) page for more information."
```

Comment (dylburger) [reply components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:32]: Changed "except" to "besides" and removed "to get" as per Google's Developer docs style guide.

File: components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      optional: true,
+    },
+    wait: {
+      type: "integer",
+      label: "Wait",
+      description: "Some websites may use javascript frameworks that may require a few extra seconds to load their content. This parameters specifies the time in milliseconds to wait for the website. Recommended values are in the interval `5000` - `10000`.",
```
Comment (dylburger) [inline components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:38]:
```suggestion
  description: "Some websites might use JavaScript frameworks that require a few extra seconds to load their content. This parameter specifies the time in milliseconds to wait for the website. Recommended values are in the interval `5000` - `10000`."
```

Comment (dylburger) [reply components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:38]: Changed "may" to "might" as per Google's Developer docs style guide.

File: components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+      optional: true,
+    },
+    waitFor: {
+      type: "string",
+      label: "Wait For",
+      description: "Specify a CSS selector and the API will wait 30 seconds until the selector appears.",
```
Comment (dylburger) [inline components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:44]:
```suggestion
  description: "Specify a CSS selector, and the API will wait 30 seconds until the selector appears."
```

Comment (dylburger) [reply components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:44]: Removed "for" as per Google's Developer docs style guide.

File: components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+      optional: true,
+    },
+    screenshot: {
+      type: "boolean",
+      label: "Screenshot",
+      description: "If you want to get a screenshot of the page you want to scrape, set screenshot parameter value to `true`.",
```
Comment (dylburger) [inline components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:50]:
```suggestion
  description: "If you want to get a screenshot of the page you'd like to scrape, set the screenshot parameter value to `true`."
```

Comment (dylburger) [reply components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:50]: Changed "you want to" to "you'd like to" as per Dreyer's English.

File: components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+      optional: true,
+    },
+    jsScenario: {
+      type: "string[]",
+      label: "JS Scenario",
+      description: "If you want to do some acts on the pages you want to scrape, for example, go to the next one, set JavaScript scenario to API call. Follow [JavaScript Execution](https://scrape-it.cloud/docs/api-features/javascript-execution) page to get more information.",
```
Comment (dylburger) [inline components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:56]:
```suggestion
  description: "If you want to perform some actions on the pages you want to scrape, for example, go to the next one, set the JavaScript scenario to the API call. Follow the [JavaScript Execution](https://scrape-it.cloud/docs/api-features/javascript-execution) page for more information."
```

Comment (dylburger) [reply components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:56]: Changed "do some acts" to "perform some actions" and removed "to get" as per Google's Developer docs style guide.

File: components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+        block_resources: blockResources,
+        block_urls: utils.parseArray(blockUrls),
+        wait,
+        wait_for: waitFor,
+        screenshot,
+        js_scenario: utils.parseArray(jsScenario)?.map(utils.parse),
+      },
+    });
+
+    step.export("$summary", `Succcessfully started scraping for ${url}`);
```
Comment (dylburger) [inline components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:87]:
```suggestion
    step.export("$summary", `Successfully started scraping for ${url}`);
```

Comment (dylburger) [reply components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:87]: Corrected the spelling of "Successfully" as per Google's Developer docs style guide.

File: components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        block_resources: blockResources,
+        block_urls: utils.parseArray(blockUrls),
+        wait,
+        wait_for: waitFor,
+        screenshot,
+        js_scenario: utils.parseArray(jsScenario)?.map(utils.parse),
+      },
+    });
+
+    step.export("$summary", `Successfully scraped ${url}.`);
```
Comment (dylburger) [inline components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:87]:
```suggestion
    step.export("$summary", `Successfully scraped ${url}.`);
```

Comment (dylburger) [reply components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:87]: No changes necessary

**Review by feyzullah — APPROVED** [review]
LGTM!

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Scape-It.cloud_6038_1042.pdf](https://github.com/PipedreamHQ/pipedream/files/11316224/Scape-It.cloud_6038_1042.pdf)

File: components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs (hunk 11 above)
Comment (dylburger) [inline components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:87]:
```suggestion
    step.export("$summary", `Successfully scraped ${url}.`);
```

Comment (dylburger) [reply components/scrape_it_cloud/actions/start-scraping/start-scraping.mjs:87]: No changes necessary

**Pull request merged**